### ISSUE TITLE
Document why KeyboardListener needs a FocusNode

### ISSUE THREAD

**Issue by xster** [flutter/flutter#116469]
The documentation could be a bit more explicit about why KeyboardListener needs a FocusNode and what is the expected interaction between them.

**Comment by justinmc** [comment]
I wonder if @gspencergoog has a definitive answer here.

**Comment by gspencergoog** [comment]
Yeah, good point.  In reality, I don't know that it really does need a focus node, since it's just a listener and doesn't affect propagation of the event.  It could just as easily register its key event listener with `HardwareKeyboard.addHandler` or `RawKeyboard.listener`.

The only practical difference is that if it has a focus node, then it can participate in the focus tree so that if a descendant handles the event before it gets there, it won't see that event.

This is a class written 7 years ago, and things have changed a bit since then...
